diff --git a/node_modules/fengari/src/lapi.js b/node_modules/fengari/src/lapi.js
index 7443fb7..fa9632f 100644
--- a/node_modules/fengari/src/lapi.js
+++ b/node_modules/fengari/src/lapi.js
@@ -161,7 +161,7 @@ const lua_xmove = function(from, to, n) {
     for (let i = 0; i < n; i++) {
         to.stack[to.top] = new lobject.TValue();
         lobject.setobj2s(to, to.top, from.stack[from.top + i]);
-        delete from.stack[from.top + i];
+        from.stack[from.top + i] = void 0;
         to.top++;
     }
 };
@@ -337,7 +337,7 @@ const lua_pushcclosure = function(L, fn, n) {
         for (let i=0; i<n; i++)
             cl.upvalue[i].setfrom(L.stack[L.top - n + i]);
         for (let i=1; i<n; i++)
-            delete L.stack[--L.top];
+            L.stack[--L.top] = void 0;
         if (n>0)
             --L.top;
         L.stack[L.top].setclCvalue(cl);
@@ -387,8 +387,8 @@ const auxsetstr = function(L, t, k) {
     api_check(L, L.top <= L.ci.top, "stack overflow");
     lvm.settable(L, t, L.stack[L.top - 1], L.stack[L.top - 2]);
     /* pop value and key */
-    delete L.stack[--L.top];
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
+    L.stack[--L.top] = void 0;
 };
 
 const lua_setglobal = function(L, name) {
@@ -418,7 +418,7 @@ const lua_setmetatable = function(L, objindex) {
         }
     }
 
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
     return true;
 };
 
@@ -426,8 +426,8 @@ const lua_settable = function(L, idx) {
     api_checknelems(L, 2);
     let t = index2addr(L, idx);
     lvm.settable(L, t, L.stack[L.top - 2], L.stack[L.top - 1]);
-    delete L.stack[--L.top];
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
+    L.stack[--L.top] = void 0;
 };
 
 const lua_setfield = function(L, idx, k) {
@@ -442,8 +442,8 @@ const lua_seti = function(L, idx, n) {
     api_incr_top(L);
     lvm.settable(L, t, L.stack[L.top - 1], L.stack[L.top - 2]);
     /* pop value and key */
-    delete L.stack[--L.top];
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
+    L.stack[--L.top] = void 0;
 };
 
 const lua_rawset = function(L, idx) {
@@ -454,8 +454,8 @@ const lua_rawset = function(L, idx) {
     let v = L.stack[L.top - 1];
     ltable.luaH_setfrom(L, o.value, k, v);
     ltable.invalidateTMcache(o.value);
-    delete L.stack[--L.top];
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
+    L.stack[--L.top] = void 0;
 };
 
 const lua_rawseti = function(L, idx, n) {
@@ -464,7 +464,7 @@ const lua_rawseti = function(L, idx, n) {
     let o = index2addr(L, idx);
     api_check(L, o.ttistable(), "table expected");
     ltable.luaH_setint(o.value, n, L.stack[L.top - 1]);
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
 };
 
 const lua_rawsetp = function(L, idx, p) {
@@ -474,7 +474,7 @@ const lua_rawsetp = function(L, idx, p) {
     let k = new TValue(LUA_TLIGHTUSERDATA, p);
     let v = L.stack[L.top - 1];
     ltable.luaH_setfrom(L, o.value, k, v);
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
 };
 
 /*
@@ -577,7 +577,7 @@ const lua_setupvalue = function(L, funcindex, n) {
         let name = aux.name;
         let val = aux.val;
         val.setfrom(L.stack[L.top-1]);
-        delete L.stack[--L.top];
+        L.stack[--L.top] = void 0;
         return name;
     }
     return null;
@@ -903,7 +903,7 @@ const lua_arith = function(L, op) {
     }
     /* first operand at top - 2, second at top - 1; result go to top - 2 */
     lobject.luaO_arith(L, op, L.stack[L.top - 2], L.stack[L.top - 1], L.stack[L.top - 2]);
-    delete L.stack[--L.top];  /* remove second operand */
+    L.stack[--L.top] = void 0;  /* remove second operand */
 };
 
 /*
@@ -946,7 +946,7 @@ const lua_setuservalue = function(L, idx) {
     let o = index2addr(L, idx);
     api_check(L, o.ttisfulluserdata(), "full userdata expected");
     o.value.uservalue.setfrom(L.stack[L.top - 1]);
-    delete L.stack[--L.top];
+    L.stack[--L.top] = void 0;
 };
 
 const checkresults = function(L,na,nr) {
@@ -1039,8 +1039,8 @@ const lua_next = function(L, idx) {
         api_incr_top(L);
         return 1;
     } else {
-        delete L.stack[L.top];
-        delete L.stack[--L.top];
+        L.stack[L.top] = void 0;
+        L.stack[--L.top] = void 0;
         return 0;
     }
 };
diff --git a/node_modules/fengari/src/ldebug.js b/node_modules/fengari/src/ldebug.js
index a471c75..13378a1 100644
--- a/node_modules/fengari/src/ldebug.js
+++ b/node_modules/fengari/src/ldebug.js
@@ -175,7 +175,7 @@ const lua_setlocal = function(L, ar, n) {
     if (local) {
         name = local.name;
         lobject.setobjs2s(L, local.pos, L.top - 1);
-        delete L.stack[--L.top];  /* pop value */
+        L.stack[--L.top] = void 0;  /* pop value */
     } else {
         name = null;
     }
diff --git a/node_modules/fengari/src/ldo.js b/node_modules/fengari/src/ldo.js
index 9c89ff7..88b3ad5 100644
--- a/node_modules/fengari/src/ldo.js
+++ b/node_modules/fengari/src/ldo.js
@@ -53,7 +53,7 @@ const adjust_top = function(L, newtop) {
             L.stack[L.top++] = new lobject.TValue(LUA_TNIL, null);
     } else {
         while (L.top > newtop)
-            delete L.stack[--L.top];
+            L.stack[--L.top] = void 0;
     }
 };
 
@@ -79,7 +79,7 @@ const seterrorobj = function(L, errcode, oldtop) {
     }
 
     while (L.top > oldtop + 1)
-        delete L.stack[--L.top];
+        L.stack[--L.top] = void 0;
 };
 
 const ERRORSTACKSIZE = LUAI_MAXSTACK + 200;
@@ -241,7 +241,7 @@ const moveresults = function(L, firstResult, res, nres, wanted) {
             for (let i = 0; i < nres; i++)
                 lobject.setobjs2s(L, res + i, firstResult + i);
             for (let i=L.top; i>=(res + nres); i--)
-                delete L.stack[i];
+                L.stack[i] = void 0;
             L.top = res + nres;
             return false;
         }
@@ -265,7 +265,7 @@ const moveresults = function(L, firstResult, res, nres, wanted) {
     }
     let newtop = res + wanted; /* top points after the last result */
     for (let i=L.top; i>=newtop; i--)
-        delete L.stack[i];
+        L.stack[i] = void 0;
     L.top = newtop;
     return true;
 };
@@ -537,7 +537,7 @@ const resume_error = function(L, msg, narg) {
     } else {
         /* remove args from the stack */
         for (let i=1; i<narg; i++)
-            delete L.stack[--L.top];
+            L.stack[--L.top] = void 0;
         lobject.setsvalue2s(L, L.top-1, ts);  /* push error message */
     }
     return LUA_ERRRUN;
diff --git a/node_modules/fengari/src/lparser.js b/node_modules/fengari/src/lparser.js
index 7c67709..e5a91d3 100644
--- a/node_modules/fengari/src/lparser.js
+++ b/node_modules/fengari/src/lparser.js
@@ -1669,7 +1669,7 @@ const luaY_parser = function(L, z, buff, dyd, name, firstchar) {
     lua_assert(!funcstate.prev && funcstate.nups === 1 && !lexstate.fs);
     /* all scopes should be correctly finished */
     lua_assert(dyd.actvar.n === 0 && dyd.gt.n === 0 && dyd.label.n === 0);
-    delete L.stack[--L.top];  /* remove scanner's table */
+    L.stack[--L.top] = void 0;  /* remove scanner's table */
     return cl;  /* closure is on the stack, too */
 };
 
diff --git a/node_modules/fengari/src/ltm.js b/node_modules/fengari/src/ltm.js
index a1aa8e3..fc310b8 100644
--- a/node_modules/fengari/src/ltm.js
+++ b/node_modules/fengari/src/ltm.js
@@ -130,7 +130,7 @@ const luaT_callTM = function(L, f, p1, p2, p3, hasres) {
 
     if (hasres) {  /* if has result, move it to its place */
         let tv = L.stack[L.top-1];
-        delete L.stack[--L.top];
+        L.stack[--L.top] = void 0;
         p3.setfrom(tv);
     }
 };
diff --git a/node_modules/fengari/src/lvm.js b/node_modules/fengari/src/lvm.js
index 423d2a7..cb4de18 100644
--- a/node_modules/fengari/src/lvm.js
+++ b/node_modules/fengari/src/lvm.js
@@ -111,12 +111,12 @@ const luaV_finishOp = function(L) {
         case OP_UNM: case OP_BNOT: case OP_LEN:
         case OP_GETTABUP: case OP_GETTABLE: case OP_SELF: {
             lobject.setobjs2s(L, base + inst.A, L.top-1);
-            delete L.stack[--L.top];
+            L.stack[--L.top] = void 0;
             break;
         }
         case OP_LE: case OP_LT: case OP_EQ: {
             let res = !L.stack[L.top - 1].l_isfalse();
-            delete L.stack[--L.top];
+            L.stack[--L.top] = void 0;
             if (ci.callstatus & lstate.CIST_LEQ) {  /* "<=" using "<" instead? */
                 lua_assert(op === OP_LE);
                 ci.callstatus ^= lstate.CIST_LEQ;  /* clear mark */
@@ -1069,7 +1069,7 @@ const luaV_concat = function(L, total) {
         total -= n - 1; /* got 'n' strings to create 1 new */
         /* popped 'n' strings and pushed one */
         for (; L.top > top-(n-1);)
-            delete L.stack[--L.top];
+            L.stack[--L.top] = void 0;
     } while (total > 1); /* repeat until only 1 result left */
 };
 
